fs-storage/src/base_storage.rs

```diff
@@ -57,6 +57,9 @@ pub trait BaseStorage<K, V>: AsRef<BTreeMap<K, V>> {
     /// from pre-configured location in the filesystem.
     fn read_fs(&mut self) -> Result<&BTreeMap<K, V>>;
 
+    /// Get a value from the internal key-value mapping.
+    fn get(&self, id: &K) -> Result<&V>;
```

Comment:
Since a BTreeMap get returns a Result it's probably best to bubble it up and let the implementation handle it. Especially so because we're implementing a very general library. The implementers can easily choose to discard the Result and treat is as an option if they want.

Reply:
I have a different perspective on this. There's a distinct difference between returning an error type and `None`. In our case of `BaseStorage`, where we're using a map with a `get` method, retrieving a value for a key can either result in finding the value (returning the value) or not finding the key (returning `None`). 

It's not considered an error if the map does not contain a value for the given key, and in Rust, it's standard practice to handle this with an `Option`.

We can see this approach in [HashMap](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get), [BTreeMap](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.get), [HashSet](https://doc.rust-lang.org/std/collections/hash_set/struct.HashSet.html#method.get), and [BTreeSet](https://doc.rust-lang.org/std/collections/struct.BTreeSet.html#method.get)

But, to accommodate future implementors of the trait who may need to handle potential errors (eg. during map access), we could consider wrapping the result in a `Result<Option<&V>>` type.

Reply:
Aah I see what you mean here about the semantics of get. In that case, `Option` is the best way to go. We don't need the additional indirection of a `Result` type because it won't add additional detail.

Also @Pushkarm029 has already implemented the `Option` in [e0b7137](https://github.com/ARK-Builders/ark-core/pull/76/commits/e0b7137db521890959e65f47d7ef0a42c20b40db) so this PR is ready to merge.